lighthouse-core/report/html/renderer/dom.js

```diff
@@ -123,13 +126,25 @@ class DOM {
       const [preambleText, linkText, linkHref] = parts.splice(0, 3);
       element.appendChild(this._document.createTextNode(preambleText));
 
+
       // Append link if there are any.
       if (linkText && linkHref) {
+        const url = new URL(linkHref);
+
+        const DEVELOPERS_GOOGLE_ORIGIN = 'https://developers.google.com';
+        if (url.origin === DEVELOPERS_GOOGLE_ORIGIN) {
+          url.searchParams.set('utm_source', 'lighthouse');
+          url.searchParams.set('utm_medium', this._lighthouseChannel);
+          if (metadata.rating) {
+            url.searchParams.set('utm_content', metadata.rating);
```

Comment:
Quoting my initial feature request:

> `utm_content` to identify why users were sent to the docs (the audit failed, passed, maybe specific details about the failure reason).

More important than simply whether the audit passed or failed, I think the more useful annotation would be why the audit failed, if we're able to extract that. For example, if there are 3 ways an audit could fail, but 90% of the docs visits are due to a single reason, that tells us a lot about what to prioritize in the docs (and says something about the state of the web).

@kaycebasques anything else that comes to mind?

Reply:
>  I think the more useful annotation would be why the audit failed, if we're able to extract that.

I don't think we have anything like that now. We do have an `explanation` sometimes, and e.g. for tap targets that would show you whether the failure was due to too close tap targets or because there's no viewport meta tag. But if we tried to use that somehow there might be a risk of leaking private data from the report though?

Reply:
It's not about passing versus failing, although that would be good to know.

The UTM parameter would let me determine whether people are coming to the documentation from Google Search, or a Lighthouse report.

If they're coming from Google Search, I can conditionally put some text at the top of the document explaining what Lighthouse is. If they're coming from Lighthouse I don't want to clutter up their experience with this redundant information.

If Lighthouse or Google Search are properly annotating their referrer values then I should be able to figure it out that way and therefore the UTM parameters aren't needed on my end. I'm just explaining the underlying motivation.

Reply:
I think we might be discussing two different things here.

@kaycebasques `utm_source=lighthouse` and `utm_medium=devtools`, `extension`, etc are good to go, so that shouldn't be a problem.

It's the `utm_content=failure` part I was talking about and wondering how useful it'll be, since it'll almost always be `failure` if it's coming from lighthouse (since a failure is the only time that link is visible by default, and someone with a failure is also a lot more interested in reading an article if they're failing something).

> More important than simply whether the audit passed or failed, I think the more useful annotation would be why the audit failed, if we're able to extract that. For example, if there are 3 ways an audit could fail, but 90% of the docs visits are due to a single reason, that tells us a lot about what to prioritize in the docs (and says something about the state of the web).

This is an interesting idea, but yeah, as @mattzeunert mentioned, we do have explanations, but they tend toward just human-readable explanations, not necessarily a systemized enumeration of failure modes, we tweak them from time to time, and they often contain report data we shouldn't share just from clicking a link (e.g. if there were conflicting canonical URLs in the page, the audit explanation includes a list of the conflicting ones).

We could explore possibilities for this back in #4663 or a separate PR?

Reply:
Agreed that `utm_content=failure` is not very helpful. I'd prefer to see that field used for the failure reason if anything. And to clarify, I don't expect it to contain any site-specific info, just a short reason code like `conflicting-canonical-urls`. If that's not feasible now then I'm ok omitting it entirely.

Reply:
Ok, let's omit it for now then. If we wanted failure reason codes we'd need to add that logic to each audit.